### DISCUSSION TITLE
[Suggestion] more bounty objective types

### DISCUSSION THREAD

**Discussion by Corvus20** [ejektaflex/Bountiful#210] [Mod Ideas]
More Bounty objective types.

Most bounties currently revolve around killing a bunch of mobs or sourcing items, but not many emphasize on more unique tasks, such as exploration or trading. I have some ideas that could provide a solution to this.

Players can accept expedition bounties for finding a certain biome or structure (and maybe additional parameters, like reaching a certain y level, harvesting certain blocks there, or spending a night there). Minecraft has a system for location-based achievements, I am unsure how easy it would be to make bounties follow this same concept, but figured I should mention this idea. Examples could include; find a [biome name] biome, find a [biome name] and harvest 20x [biome related resource], find a [structure name]. find a [structure name] and kill 5x [structure related mob].

Players could accept merchant bounties for trading with villagers at a village indicated via coordinates, and possibly even try to get certain items to fill out the bounty's completion requirements. Making trade runs could be a cool addition to a game. Examples of this could be; go to the village at (X coordinates, Y coordinates) and make 5 trades, go to the village at (X coordinates, Y coordinates) and trade to get 32x (item name).

Defense bounties will trigger a raid or a spawning of a bunch of mobs nearby when accepted. The victory condition is defeating the raid or killing the mobs spawned by the bounty. If combined with a mod like "Savage and Ravage" or the "Guard Villagers"
 mod, participating in village defense can be a far more interesting experience.

Animal Catching bounties will be completed by having a player attach a lead to an animal (or animals) specified by the quest. This will make the animals disappear, similar to how quest items would in other bounty types. Perhaps compatibility with other mob capture tools from other mods (ender leads, morbs, nets, etc.) could be implemented as well for capturing hostile mobs, or other mobs that are not usually able to have a lead tied to them.

I know I just made another suggestion post, but I wanted to keep this separate to not be too overwhelming. Sorry if I am clogging things up.

**Comment by ejektaflex** [comment]
These are some neat ideas. Finding biomes would be fun - That would probably be one of the most likely things to get added, but does suffer from the fact that as you play longer, those bounties would get harder to complete 🤔 .

**Comment by Corvus20** [comment]
Perhaps, yeah. Maybe make it so that as you complete more of them, the reward goes up to scale with that difficulty? Idk if this would be practical, but its an idea at least.

**Comment by ejektaflex** [comment]
Finding new biomes / structures would be difficult to track with code in such a way that you can't keep re-finding the same places.

However, I do like the idea of making villager trades a completion requirement!

Same for animal catching. Bringing back an animal on a lead could be some fun. It would just have to have enough variety that you can't create a farm and farm bounties with it.

**Comment by ejektaflex** [comment]
I've shifted my priorities around a lot - and as such, I don't think I'm going to be working on any new features for Bountiful-Forge for a little while. All of my focus is on a bigger, better version for Fabric. Yet, I still love these suggestions, so I'll be transferring this issue to that repository. Thank you!
